src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private TruckProducer truckProducer = new TruckProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Truck":
+                return new ArrayList<>(truckProducer.get());
+            case "Excavator":
+                return new ArrayList<>(excavatorProducer.get());
+            case "Bulldozer":
+                return new ArrayList<>(bulldozerProducer.get());
+            default:
+                return new ArrayList<>();
```

Comment:
Also I'd recommend creating a map as a class field (using class as a key and producer as a value) and using this map instead of switch